iosxr_iso2vbox.py

```diff
@@ -608,45 +591,122 @@ def main(argv):
     with tarfile.open(box_tmp, 'a') as tarf:
         tarf.add("./metadata.json")
     run(['gzip', '--force', '-S', '.box', box_tmp])
+    # gzip automatically cleans up - no need for os.remove(box_tmp)
 
     logger.info('Created: %s', box_out)
+    return box_out
 
-    # Create OVA
-    if create_ova is True:
-        logger.info('Creating OVA %s', ova_out)
-        run(['VBoxManage', 'export', vmname, '--output', ova_out])
-        logger.debug('Created OVA %s', ova_out)
 
-    # Clean up VM used to generate box
-    cleanup_vmname(vmname, vbox)
+def vbox_to_ova(vmname, box_dir):
+    """Export the VirtualBox VM to OVA format."""
+    ova_out = os.path.join(box_dir, vmname + '.ova')
+
+    # Delete existing OVA
+    if os.path.exists(ova_out):
+        os.remove(ova_out)
+        logger.debug('Found and deleted previous %s', ova_out)
+
+    logger.info('Creating OVA %s', ova_out)
+    run(['VBoxManage', 'export', vmname, '--output', ova_out])
+    logger.debug('Created OVA %s', ova_out)
+
+
+def main():
+    """Main function."""
+    input_iso = ''
+    create_ova = False
+
+    args = parse_args()
+
+    # Handle Input ISO (Local or URI)
+    if re.search(':/', args.ISO_FILE):
+        # URI Image
+        cmd_string = 'scp %s@%s .' % (getpass.getuser(), args.ISO_FILE)
+        logger.debug('Will attempt to scp the remote image to current working dir. You may be required to enter your password.')
+        logger.debug('%s\n', cmd_string)
+        subprocess.call(cmd_string, shell=True)
+        input_iso = os.path.basename(args.ISO_FILE)
+    else:
+        # Local image
+        input_iso = args.ISO_FILE
+
+    # Handle create OVA
+    create_ova = args.create_ova
+
+    if not os.path.exists(input_iso):
+        sys.exit('%s does not exist' % input_iso)
+
+    # Set Virtualbox VM name from the input ISO
+    vmname = os.path.basename(os.path.splitext(input_iso)[0])
+
+    set_logging()
+    logger.setLevel(level=args.verbose)
+
+    # Set up paths
+    base_dir = os.path.join(os.getcwd(), 'machines')
+
+    logger.debug('Input ISO is %s', input_iso)
+    logger.debug('VM Name:  %s', vmname)
+    logger.debug('base_dir: %s', base_dir)
+
+    vbox = create_vbox_vm(vmname, base_dir, input_iso)
+
+    box_dir = os.path.dirname(vbox)
+
+    try:
+        launch_and_configure_vbox_vm(vmname, box_dir, args.verbose, args.debug)
+
+        box_out = vbox_to_vagrant(vmname, box_dir)
+
+        # Create OVA
+        if create_ova is True:
+            vbox_to_ova(vmname, box_dir)
+    finally:
+        # Attempt to clean up after ourselves even if something went wrong
+        cleanup_vmname(vmname, vbox)
 
     # Run basic sanity tests unless -s
     if not args.skip_test:
         logger.info('Running basic unit tests on Vagrant VirtualBox...')
 
-        if args.verbose == logging.DEBUG:
-            verbose_str = '-v'
-        else:
-            # Shhhh...
-            verbose_str = ''
+        # hackety hack hack hack...
```

Comment:
I recall from Python 101 class, that there was a way to add a file to a python repo that allows all files in that repo to be imported. I know I wanted to investigate this more. Maybe future enhancement?